src/Features/Core/Portable/Diagnostics/Service/DiagnosticAnalyzerService.cs

```diff
@@ -54,6 +73,10 @@ public DiagnosticAnalyzerService(
                 RequestDiagnosticRefresh();
             }
         });
+
+        // When the workspace changes what context a document is in (when a user picks a different tfm to view the
+        // document in), kick off a refresh so that diagnostics properly update in the task list and editor.
+        workspace.RegisterDocumentActiveContextChangedHandler(args => RequestDiagnosticRefresh());
```

Comment:
moved registration here, which is much cleaner than when we were originally creating the incremetnal analyzers.